app/models.py

```diff
@@ -30,6 +32,8 @@ class Service(db.Model):
                                nullable=False)
     data = db.Column(JSON)
 
+    status = db.Column(db.String, index=False, unique=False, nullable=False)
```

Comment:
To your second point. I would expect the database to handle optimisation of low cardinality columns for me. Caveat: I have done no research to verify that Postgres actually does this.